.github/workflows/fhirr4-build-executor.yml

```diff
@@ -0,0 +1,158 @@
+        if: inputs.bal_central_environment == 'DEV'
+        uses: ballerina-platform/ballerina-action@2201.7.0
+        with:
+          args:
+            push
+        env:
+          WORKING_DIR: /fhirr4/ballerina/target/classes/fhirservice
+          JAVA_HOME: /usr/lib/jvm/default-jvm
+          BALLERINA_DEV_CENTRAL: true
+          BALLERINA_CENTRAL_ACCESS_TOKEN: ${{ secrets.BALLERINA_CENTRAL_DEV_ACCESS_TOKEN }}     
+
+      - name: Push to Prod
+        if: inputs.bal_central_environment == 'PROD'
+        uses: ballerina-platform/ballerina-action@2201.7.0
+        with:
+          args:
+            push
+        env:
+          WORKING_DIR: /fhirr4/ballerina/target/classes/fhirservice
+          JAVA_HOME: /usr/lib/jvm/default-jvm
+          BALLERINA_CENTRAL_ACCESS_TOKEN: ${{ secrets.BALLERINA_CENTRAL_ACCESS_TOKEN }}
+
+      - name: Publish Release
+        if: inputs.bal_central_environment == 'PROD'
+        id: publish_release
+        run: |
+          # Get Branch Name
+          BRANCH_NAME=$(echo ${GITHUB_REF#refs/heads/})
+          echo "BRANCH_NAME=${BRANCH_NAME}" >> $GITHUB_OUTPUT
+          # Release name
+          RELEASE_NAME=${BRANCH_NAME#release-}
+          curl \
+            -s -X POST \
+            -H "Authorization: Bearer ${{ secrets.BALLERINA_BOT_TOKEN }}" \
+            -H "Accept: application/vnd.github.v3+json" \
+            -d '{
+              "tag_name": "'$RELEASE_NAME'",
+              "name": "'$RELEASE_NAME'",
+              "body": "[Automated] Creating tag:  '$RELEASE_NAME'.",
+              "draft": false,
+              "prerelease": false,
+              "target_commitish": "'$BRANCH_NAME'"
+            }' \
+            "https://api.github.com/repos/${{ github.repository }}/releases"
+
+      - name: Update version in the pom files
+        if: ${{ inputs.bal_central_environment == 'PROD' }}
+        id: increment_patch_version
+        run: |
+          CURRENT_VERSION=$(grep -Po -m 1 '(?<=<version>)[\d.]+' ${{ inputs.working_dir }}/pom.xml)
+          IFS='.' read -ra VERSION_PARTS <<< "$CURRENT_VERSION"
+          PATCH_VERSION=$((VERSION_PARTS[2] + 1))
+          NEW_VERSION="${VERSION_PARTS[0]}.${VERSION_PARTS[1]}.$PATCH_VERSION"
+          echo "NEW_VERSION=${NEW_VERSION}" >> $GITHUB_OUTPUT
+          echo "CURRENT_VERSION=${CURRENT_VERSION}" >> $GITHUB_OUTPUT
+          #Bump the versions in each pom file in fhirr4
+          sed -i "0,/<version>${CURRENT_VERSION}<\/version>/s//<version>${NEW_VERSION}<\/version>/" ${{ inputs.working_dir }}/pom.xml
+          sed -i "0,/<version>${CURRENT_VERSION}<\/version>/s//<version>${NEW_VERSION}<\/version>/" ${{ inputs.working_dir }}/compiler-plugin/pom.xml
+          sed -i "0,/<version>${CURRENT_VERSION}<\/version>/s//<version>${NEW_VERSION}<\/version>/" ${{ inputs.working_dir }}/native/pom.xml
+          sed -i "0,/<version>${CURRENT_VERSION}<\/version>/s//<version>${NEW_VERSION}<\/version>/" ${{ inputs.working_dir }}/ballerina/pom.xml
```

Comment:
This is fine for now. But as a future improvement let's use an XML processing tool to do this.

Reply:
Got it👍